src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === null || this[i] === undefined) && i === 0) {
+        continue;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        newString += separator;
+        continue;
+      }
+
+      if (i === 0) {
+        newString += this[i];
+        continue;
+      }
+      newString += `${separator}${this[i]}`;
```

Comment:
instead of handling all other cases, we can try to look for valid ones and then perform our operation, that way we can get rid of multiple continue statements:
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
           // add letter to newString
       }
      
      if(iterator is less than word length) // add sepatator to newString
```